src/dart/lib/widgets.dart

```diff
@@ -0,0 +1,288 @@
+  const IconVG(this.asset, {
+    Key? key,
+  }) : super(key: key);
+
+  /// The name of the asset to render.
+  ///
+  /// The ambient [DefaultAssetBundle] is used to resolve this name.
+  final String asset;
+
+  @override
+  Widget build(BuildContext context) {
+    final IconThemeData theme = IconTheme.of(context);
+    return Opacity(
+      opacity: theme.opacity!,
+      child: IconVGImage(
+        asset,
+        palette: <Color>[ theme.color! ],
+        size: Size.square(theme.size!),
+      ),
+    );
+  }
+}
+
+/// Render the specified IconVG [asset].
+///
+/// If the image is not available in the application's assets, consider fetching
+/// the bytes manually and using a [RawIconVGImage].
+class IconVGImage extends StatefulWidget {
+  const IconVGImage(this.asset, {
+    Key? key,
+    this.palette,
+    this.size,
+    this.clipBehavior = Clip.hardEdge,
+  }) : super(key: key);
+
+  /// The name of the asset to render.
+  ///
+  /// The ambient [DefaultAssetBundle] is used to resolve this name.
+  final String asset;
+
+  /// The custom color palette used to decode the image.
+  ///
+  /// Must not have more than 64 colors.
+  final List<Color>? palette;
+
+  /// The target render size. Defaults to the [BoxConstraints.maxHeight].
+  final Size? size;
+
+  /// How to clip the image.
+  ///
+  /// This can be used to enable anti-aliased clips if the image is rotated, or
+  /// disable clips entirely (for improved performance) if the image does not
+  /// need them.
+  final Clip clipBehavior;
+
+  @override
+  State<IconVGImage> createState() => _IconVGImageState();
+}
+
+class _IconVGImageState extends State<IconVGImage> {
+  AssetBundle? _assetBundle;
+  Uint8List? _bytes;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final AssetBundle assetBundle = DefaultAssetBundle.of(context);
+    if (assetBundle != _assetBundle) {
+      _assetBundle = assetBundle;
+      _bytes = null;
+      _fetchBytes();
+    }
+  }
+
+  @override
+  void didUpdateWidget(IconVGImage oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.asset != widget.asset) {
+      _bytes = null;
+      _fetchBytes();
+    }
+  }
+
+  Future<void> _fetchBytes() async {
+    assert(_assetBundle != null);
+    final ByteData data = await _assetBundle!.load(widget.asset);
+    if (mounted) {
+      setState(() {
+        _bytes = data.buffer.asUint8List();
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return RawIconVGImage(
+      bytes: _bytes,
+      palette: widget.palette,
+      size: widget.size,
+      clipBehavior: widget.clipBehavior,
+    );
+  }
+}
+
+/// Render the specified [bytes] as an IconVG image.
+///
+/// If the image is an asset, consider using [IconVGImage] instead.
+class RawIconVGImage extends StatefulWidget {
+  const RawIconVGImage({
+    Key? key,
+    this.bytes,
+    this.palette,
+    this.size,
+    this.clipBehavior = Clip.hardEdge,
```

Comment:
Consider adding a `semanticsLabel` and bubbling that up/down.

Reply:
Done.